### PR TITLE
Define some special error for k8s status error

### PR DESCRIPTION
**Pull request by yuzp1996**
For example FileNotFound GitRevisionNotFound etc.

# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [`spec` PR link](https://github.com/katanomi/spec) included
- [ ] Follows the [commit message standard](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md#commits)
- [ ] Meets the [contributing guidelines](https://github.com/katanomi/spec/blob/main/CONTRIBUTING.md) (including
  functionality, content, code)
- [ ] Test cases with documentation and functionality works as expected using current and related github repos (MUST deploy and check)
- [ ] Release notes block below has been filled in or deleted (only if no user facing changes)

# Release Notes

### CODE REVIEW

File: errors/errorhandling.go [hunk 1]
```diff
@@ -66,6 +78,12 @@ func AsStatusError(response *resty.Response, grs ...schema.GroupResource) error
 		false,
 	)
 
+	// if the response is a metav1.status use it's reason directly
+	var status metav1.Status
+	if json.Unmarshal([]byte(response.String()), &status) == nil && status.Reason != "" {
```
Comment (yuzp1996) [inline errors/errorhandling.go:83]: reason 目前只能从 response 中解析成 metav1.S tatus 然后再读取 Reason

File: errors/errorhandling.go [hunk 2]
```diff
@@ -47,6 +48,17 @@ func AsStatusCode(err error) int {
 	return http.StatusInternalServerError
 }
 
+// HandleError handles error in requests
+func HandleError(req *restful.Request, resp *restful.Response, err error) {
+	err = AsAPIError(err)
+	status := AsStatusCode(err)
+	if statusErr, ok := err.(errors.APIStatus); ok {
```
Comment (yuzp1996) [inline errors/errorhandling.go:55]: 代码位置改动

如果错误是 errors.APIStatus 的错误，则直接将信息写入到 response body 中了，

**Review by danielfbm — APPROVED** [review]

File: errors/convert_response_error.go [hunk 3]
```diff
@@ -24,8 +24,39 @@ import (
...
+	var httpres *http.Response
+	if response != nil {
+		httpres = response
+	}
+	return ConvertResponseError(ctx, httpres, err, gvk, names...)
+}
+
+func commonRespError(response *http.Response, err error) (isCommonError bool, commonError error) {
+	if response != nil && response.StatusCode == http.StatusServiceUnavailable {
+		return true, NewToolServiceUnavailable(err.Error())
```
Comment (nanjingfm) [inline errors/convert_response_error.go:50]: 为啥是 toolServiceUnavailable  而不是 ServiceUnavaliable?

这个方法应该不是特指请求工具插件或者工具 api 的吧？

Comment (yuzp1996) [reply errors/convert_response_error.go]: 目前已经更新到了新的文件中并且修改了相关函数的名称

**Review by nanjingfm — DISMISSED** [review]

**Review by nanjingfm — APPROVED** [review]

**Pull request merged**